### PR TITLE
Remove the restriction on sum of memory_spill_ratio and memory_shared_quota.

### PR DESCRIPTION
**Pull request by guofengrichard**
This commit does two changes:
1. Remove the restriction that sum of memory_spill_ratio and memory_shared_quota
must be no larger than 100.
2. Change the range of memory_spill_ratio to be [0, 100].

### CODE REVIEW

File: src/backend/commands/resgroupcmds.c [hunk 1]
```diff
@@ -507,14 +506,13 @@ AlterResourceGroup(AlterResourceGroupStmt *stmt)
 			if (memSpillRatioNew < RESGROUP_MIN_MEMORY_SPILL_RATIO)
 				ereport(ERROR,
 						(errcode(ERRCODE_INVALID_LIMIT_VALUE),
-						 errmsg("memory spill ratio must be greater than %d",
+						 errmsg("memory spill ratio must be no less than %d",
```
Comment (pivotal-ning-yu) [inline src/backend/commands/resgroupcmds.c]: For concurrency we have "concurrency limit cannot be less than %d" as the error message, would it be better to use a consistent format for memory spill.

File: src/test/isolation2/sql/resgroup/resgroup_unlimit_memory_spill_ratio.sql [hunk 2]
```diff
@@ -0,0 +1,62 @@
+-- start_ignore
```
Comment (xiong-gang) [inline src/test/isolation2/sql/resgroup/resgroup_unlimit_memory_spill_ratio.sql]: can we move the tests in resgroup/resgroup_zero_workmem into this file?

File: src/backend/commands/resgroupcmds.c (hunk 1 above)
Comment (guofengrichard) [reply src/backend/commands/resgroupcmds.c]: Make sense. Will format the messages.

File: src/test/isolation2/sql/resgroup/resgroup_unlimit_memory_spill_ratio.sql (hunk 2 above)
Comment (guofengrichard) [reply src/test/isolation2/sql/resgroup/resgroup_unlimit_memory_spill_ratio.sql]: I think it is better to keep 'resgroup_zero_workmem' as a separate test and re-write the test to create the resource group with memory_spill_ratio set to zero.

**Pull request merged**